### ISSUE TITLE
Train VMamba2 but gradients turned into NaN

### ISSUE THREAD

**Issue by Mark2Lutz** [MzeroMiko/VMamba#300]
Hi, thanks for your great work, I set up the training with vmamba2, but after a period of time, the gradients turned into NaN. Even with gradient clipping, NaN still occurs. Have you encountered this situation before, and do you have any good solutions?

**Comment by MzeroMiko** [comment]
If you are talking about VMamba with mamba2, then I have not trained it due to it's slow training on my device.

If you are talking about VMamba with SS2Dv2, then I have not encountered the NaN problem before.

**Comment by leenas233** [comment]
> Hi, thanks for your great work, I set up the training with vmamba2, but after a period of time, the gradients turned into NaN. Even with gradient clipping, NaN still occurs. Have you encountered this situation before, and do you have any good solutions?

what task did you try? it happened to me as well in the detection task. i guess the reason is the long sequence when training

**Comment by poult-lab** [comment]
I used the VMamba architecture (not VMamba2) for ECG classification and encountered NaN issues in the gradients.
